angr/storage/memory_mixins/regioned_memory/region.py

```diff
@@ -0,0 +1,166 @@
+from typing import Optional, TYPE_CHECKING
+import copy
+
+from .. import MemoryMixin
+
+if TYPE_CHECKING:
+    from .. import RegionMemory
+
+
+class MemoryRegion:
+
+    __slots__ = ('_endness', '_id', '_state', '_is_stack', '_related_function_addr', '_alocs', '_memory', )
+
+    def __init__(self, mem_id, state, related_function_addr=None, init_memory=True, cle_memory_backer=None,
+                 dict_memory_backer=None, endness=None):
+        self._endness = endness
+        self._id = mem_id
+        self._state = state
+        self._is_stack = mem_id.startswith('stack_') # TODO: Fix it
+        self._related_function_addr = related_function_addr
+        # This is a map from tuple (basicblock_key, stmt_id) to
+        # AbstractLocation objects
+        self._alocs = { }
+
+        self._memory: Optional['RegionMemory'] = None
+
+        if init_memory:
+            # delayed import
+            from .. import RegionMemory
+            self._memory = RegionMemory(memory_id=mem_id, endness=self._endness,
+                                        cle_memory_backer=cle_memory_backer, dict_memory_backer=dict_memory_backer,
+                                        )
+
+            self._memory.set_state(state)
+
+    @property
+    def id(self):
+        return self._id
```

Comment:
why do you do this. why not just make the attributes public. we do not need to enforce things being read-only